src/pages/DistributePage/queries.ts

```diff
@@ -85,3 +90,38 @@ export function useGetAllocations(epochId: number) {
     });
   });
 }
+
+export const getPreviousDistribution = async (
+  circle_id: number
+): Promise<typeof distributions | undefined> => {
+  const { distributions } = await client.query({
+    distributions: [
+      {
+        order_by: [{ id: order_by.desc }],
+        where: {
+          epoch: { circle_id: { _eq: circle_id } },
+          saved_on_chain: { _eq: true },
+        },
+      },
+      {
+        id: true,
+        vault_id: true,
+        distribution_json: [{}, true],
+      },
+    ],
+  });
+  return distributions;
+};
+
+export type PreviousDistribution = Exclude<
+  Awaited<ReturnType<typeof getPreviousDistribution>>,
+  undefined
+>[0];
+
+export function usePreviousDistributions(circleId: number) {
```

Comment:
Should allow nullish values here since it handles them with `enabled`

```suggestion
export function usePreviousDistributions(circleId: number | null | undefined) {
```

Reply:
good catch